cmd/juju/application/upgradecharm.go

```diff
@@ -220,9 +218,6 @@ func (c *upgradeCharmCommand) SetFlags(f *gnuflag.FlagSet) {
 func (c *upgradeCharmCommand) Init(args []string) error {
 	switch len(args) {
 	case 1:
-		if !names.IsValidApplication(args[0]) {
```

Comment:
Hey, sorry, I forgot that this was the case (I even typed out my QA steps wrong...). You are correct that this is the application name, but the check still shouldn't be done client-side (see #2 in commit message)